src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -26,18 +27,18 @@ import { lidoService } from '@/services/lido/lido.service';
 const IS_LIQUIDITY_MINING_ENABLED = true;
 
 export default class Pools {
-  service: Service;
+  service: BalancerSubgraphService;
   query: QueryBuilder;
   networkId: NetworkId;
 
   constructor(
-    service: Service,
+    service: BalancerSubgraphService,
     query: QueryBuilder = queryBuilder,
-    private readonly configService = _configService
+    private readonly configService = Container.get(ConfigService)
   ) {
     this.service = service;
     this.query = query;
-    this.networkId = Number(configService.env.NETWORK) as NetworkId;
+    this.networkId = Number(this.configService.env.NETWORK) as NetworkId;
```

Comment:
Since we're only accessing this instance passed inside the constructor and it's passed in as a param, what's the need to use `this` here?

Reply:
Good catch! When using `TypeDI` I prefer using `this.`, even in the constructor, to indicate that the service you are passing in is actually accessible on `this` (as apposed to having to be manually assigned in the constructor). This is useful if the class implements any methods so you know you already have access to `this.configService`. Again, mostly personal preference.

In this case `Pools` does not implement the `@Service` decorator - making it necessary to use the `Container.get(...)` syntax. Ideally all classes in `/services` would be decorated with `@Service` which would eliminate the concept of `passing as a param` in exchange for `injecting a service` that one has access to using `this`.